### ISSUE TITLE
doesnt working for every option of input features

### ISSUE THREAD

**Issue by EnderMagic** [rsxdalv/tts-generation-webui#125]
is my first time use it
![微信图片_20230813155425](https://github.com/rsxdalv/tts-generation-webui/assets/31653737/fba6e9ec-75ce-4aa5-b2ed-940dbf312277)

`Package                   Version
------------------------- ---------
absl-py                   1.4.0
accelerate                0.21.0
aiofiles                  23.2.1
aiohttp                   3.8.5
aiosignal                 1.3.1
albumentations            1.3.1
altair                    5.0.1
annotated-types           0.5.0
antlr4-python3-runtime    4.8
anyio                     3.7.1
appdirs                   1.4.4
async-timeout             4.0.3
attrs                     23.1.0
audiocraft                1.0.0
audiolm-pytorch           1.1.4
audioread                 3.0.0
av                        10.0.0
bark-hubert-quantizer     0.0.4
beartype                  0.15.0
bitarray                  2.8.1
blis                      0.7.10
boto3                     1.28.25
botocore                  1.31.25
brotlipy                  0.7.0
cachetools                5.3.1
catalogue                 2.0.9
certifi                   2023.7.22
cffi                      1.15.1
charset-normalizer        2.0.4
click                     8.1.6
cloudpickle               2.2.1
colorama                  0.4.6
colorlog                  6.7.0
confection                0.1.1
contourpy                 1.1.0
cryptography              41.0.2
cycler                    0.11.0
cymem                     2.0.7
Cython                    0.29.36
decorator                 5.1.1
demucs                    4.0.0
diffq                     0.2.4
docopt                    0.6.2
dora-search               0.1.12
einops                    0.6.1
ema-pytorch               0.2.3
encodec                   0.1.1
exceptiongroup            1.1.2
fairseq                   0.12.4
faiss-cpu                 1.7.4
fastapi                   0.101.0
ffmpeg-python             0.2.0
ffmpy                     0.3.1
filelock                  3.9.0
flashy                    0.0.2
fonttools                 4.42.0
frozenlist                1.4.0
fsspec                    2023.6.0
functorch                 2.0.0
funcy                     2.0
future                    0.18.3
google-auth               2.22.0
google-auth-oauthlib      1.0.0
gradio                    3.35.2
gradio_client             0.4.0
grpcio                    1.57.0
h11                       0.14.0
httpcore                  0.17.3
httpx                     0.24.1
huggingface-hub           0.16.4
hydra-colorlog            1.2.0
hydra-core                1.0.7
idna                      3.4
inflect                   7.0.0
insightface               0.7.3
Jinja2                    3.1.2
jmespath                  1.0.1
joblib                    1.3.1
json5                     0.9.14
jsonschema                4.19.0
jsonschema-specifications 2023.7.1
julius                    0.2.7
kiwisolver                1.4.4
lameenc                   1.6.1
langcodes                 3.3.0
lazy_loader               0.3
librosa                   0.9.2
lightning-utilities       0.9.0
linkify-it-py             2.0.2
lion-pytorch              0.1.2
llvmlite                  0.39.0
local-attention           1.8.6
lxml                      4.9.3
Markdown                  3.4.4
markdown-it-py            2.2.0
MarkupSafe                2.1.1
matplotlib                3.7.2
matplotlib-inline         0.1.6
mdit-py-plugins           0.3.3
mdurl                     0.1.2
mkl-fft                   1.3.6
mkl-random                1.2.2
mkl-service               2.4.0
mpmath                    1.3.0
msgpack                   1.0.5
multidict                 6.0.4
murmurhash                1.0.9
mypy-extensions           1.0.0
networkx                  3.1
num2words                 0.5.12
numba                     0.56.4
numpy                     1.23.5
oauthlib                  3.2.2
omegaconf                 2.0.6
onnx                      1.14.0
opencv-python-headless    4.8.0.74
openunmix                 1.2.1
orjson                    3.9.4
packaging                 23.1
pandas                    2.0.3
pathy                     0.10.2
Pillow                    9.3.0
pip                       23.2.1
pooch                     1.6.0
portalocker               2.7.0
praat-parselmouth         0.4.3
preshed                   3.0.8
progressbar               2.5
protobuf                  4.24.0
psutil                    5.9.5
pyasn1                    0.4.8
pyasn1-modules            0.2.8
pycparser                 2.21
pydantic                  1.10.12
pydantic_core             2.4.0
pydub                     0.25.1
Pygments                  2.16.1
pyOpenSSL                 23.2.0
pyparsing                 3.0.9
pyre-extensions           0.0.29
PySocks                   1.7.1
python-dateutil           2.8.2
python-dotenv             1.0.0
python-multipart          0.0.6
pytz                      2023.3
pywin32                   306
pyworld                   0.3.4
PyYAML                    6.0.1
qudida                    0.0.4
referencing               0.30.2
regex                     2023.8.8
requests                  2.31.0
requests-oauthlib         1.3.1
resampy                   0.4.2
retrying                  1.3.4
rotary-embedding-torch    0.2.7
rpds-py                   0.9.2
rsa                       4.9
rvc-beta                  0.1.1
s3transfer                0.6.1
sacrebleu                 2.3.1
safetensors               0.3.2
scikit-learn              1.3.0
scipy                     1.9.3
semantic-version          2.10.0
sentencepiece             0.1.99
setuptools                68.0.0
six                       1.16.0
smart-open                6.3.0
sniffio                   1.3.0
soundfile                 0.12.1
soxr                      0.3.5
spacy                     3.5.2
spacy-legacy              3.0.12
spacy-loggers             1.0.4
srsly                     2.4.7
starlette                 0.27.0
submitit                  1.4.5
suno-bark                 0.0.1a0
sympy                     1.11.1
tabulate                  0.9.0
tensorboard               2.14.0
tensorboard-data-server   0.7.1
tensorboard-plugin-wit    1.8.1
tensorboardX              2.6.2
thinc                     8.1.12
threadpoolctl             3.2.0
tokenizers                0.13.3
toolz                     0.12.0
torch                     2.0.0
torchaudio                2.0.0
torchcrepe                0.0.20
torchgen                  0.0.1
torchmetrics              1.0.3
torchvision               0.15.0
tornado                   6.3.3
TorToiSe                  2.6.0
tqdm                      4.66.1
traitlets                 5.9.0
transformers              4.31.0
treetable                 0.2.5
typer                     0.7.0
typing_extensions         4.7.1
typing-inspect            0.9.0
tzdata                    2023.3
uc-micro-py               1.0.2
Unidecode                 1.3.6
urllib3                   1.26.16
uvicorn                   0.21.1
vector-quantize-pytorch   1.6.30
vocos                     0.0.2
wasabi                    1.1.2
websockets                11.0.3
Werkzeug                  2.3.6
wheel                     0.38.4
win-inet-pton             1.1.0
xformers                  0.0.19
yarl                      1.9.2`

**Comment by EnderMagic** [comment]
![微信图片_20230813155806](https://github.com/rsxdalv/tts-generation-webui/assets/31653737/a6ca7de1-bf71-4591-bf55-295da03c54e9)

**Comment by rsxdalv** [comment]
Thanks for reporting this!
First thing to note - deepspeed _basically_ does not work on widnows (despite Microsoft making deepspeed)
Also - did you try installing custom models? I'm wondering why there's an error for the model selection, I couldn't replicate it so I must be missing something.

**Comment by EnderMagic** [comment]
No it is original.and i also try import model but still doesn't work.i also installed deepspeed0.8.3

**Comment by rsxdalv** [comment]
I'll test again a clean install on colabs to see if it works. Additionally there was a pytorch reinstall issue that should be resolved now.

**Comment by EnderMagic** [comment]
> I'll test again a clean install on colabs to see if it works. Additionally there was a pytorch reinstall issue that should be resolved now.

Is not single environment that is using my system environment?my python is 3.10.8 with pytorch2.0.1 cuda11.8

**Comment by rsxdalv** [comment]
If you installed using conda then it should be it's own virtual environment. The path in your screenshots suggests that you might have, but I'm certain.
If you installed it together with something else it's likely that gradio/gradio_client packages are not matching and it _just breaks_. I had an issue like that before. You could probably fix it but I go with a "clean, isolated install" approach.

**Comment by EnderMagic** [comment]
I did still same problem

**Comment by rsxdalv** [comment]
Strange, it works on colab.
Gradio versions are OK.
Torch seems OK (although project uses 2.0.0, it _probably_ works with 2.0.1), and if it did it should fail differently.
URL is correct in both screenshots.
Honestly the specs all seem to match up. I'm a little bit confused that it says oobabooga in the install path, but that's _probably_ fine.
(For reference, this is what my path is like:
```
C:\Users\admin\Desktop\one-click-installers-tts-main\tts-generation-webui\src\rvc_tab\rvc_tab.py
```
)

**Comment by EnderMagic** [comment]
> 奇怪的是，它可以在 colab 上运行。 渐变版本还可以。 Torch 看起来没问题（虽然项目使用 2.0.0，但它_可能_适用于 2.0.1），如果确实如此，它应该以不同的方式失败。 两个屏幕截图中的 URL 都是正确的。 老实说，所有规格似乎都匹配。我有点困惑，它在安装路径中显示 oobabooga，但这_可能_没问题。 （作为参考，这就是我的路径：
> 
> ```
> C:\Users\admin\Desktop\one-click-installers-tts-main\tts-generation-webui\src\rvc_tab\rvc_tab.py
> ```
> 
> ）

![微信图片_20230814025409](https://github.com/rsxdalv/tts-generation-webui/assets/31653737/98e92bf7-f0b2-4f11-9906-23b0308bbf2c)
![微信图片_20230814025419](https://github.com/rsxdalv/tts-generation-webui/assets/31653737/d37cf8b0-7b59-48c5-be75-52a12db8f808)

**Comment by EnderMagic** [comment]
(D:\one-click-installers-tts-6.0\installer_files\env) D:\one-click-installers-tts-6.0>python -c "import torch; print('torch:', torch.__version__, torch)"
torch: 2.0.0 <module 'torch' from 'D:\\one-click-installers-tts-6.0\\installer_files\\env\\lib\\site-packages\\torch\\__init__.py'>

(D:\one-click-installers-tts-6.0\installer_files\env) D:\one-click-installers-tts-6.0>python -c "import torch; print('CUDA available:', torch.cuda.is_available())"
CUDA available: True

(D:\one-click-installers-tts-6.0\installer_files\env) D:\one-click-installers-tts-6.0>

**Comment by EnderMagic** [comment]
![微信图片_20230814031526](https://github.com/rsxdalv/tts-generation-webui/assets/31653737/373b3c4b-81b3-44a6-becd-86f556280a83)
It is stuck, should I use the gpu version of colab?

**Comment by rsxdalv** [comment]
It's not stuck, CPU is very slow for Tortoise. I recommend the "T4" GPU
option.

**Comment by rsxdalv** [comment]
I will close this issue, if there's more input, please let me know.

As for the deepspeed/Windows, I think there should be options to install it now, or in the not so distant future.
